sdk/core/core-rest-pipeline/test/node/formDataPolicy.spec.ts

```diff
@@ -37,4 +37,102 @@ describe("formDataPolicy (node-only)", function () {
     const content = Buffer.concat(buffers);
     assert.deepEqual([...content], [...Buffer.from("aaa")]);
   });
+
+  it("prepares a form with multiple fields correctly", async function () {
+    // add field with spooky unicode characters to ensure encoding is working
+    const result = await performRequest({ a: "va", b: "vb", c: "👻👻" });
+    assert.isUndefined(result.request.formData);
+    const multipartBody = result.request.multipartBody as any;
+    assert.ok(multipartBody, "expecting multipartBody to be defined");
+    const parts = (multipartBody as any).parts as BodyPart[];
+    const enc = new TextEncoder();
+    assert.ok(parts.length === 3, "need 3 parts");
+    assert.deepEqual(parts[0], {
+      headers: createHttpHeaders({
+        "Content-Disposition": `form-data; name="a"`,
+      }),
+      body: enc.encode("va"),
+    });
+    assert.deepEqual(parts[1], {
+      headers: createHttpHeaders({
+        "Content-Disposition": `form-data; name="b"`,
+      }),
+      body: enc.encode("vb"),
+    });
+    assert.deepEqual(parts[2], {
+      headers: createHttpHeaders({
+        "Content-Disposition": `form-data; name="c"`,
+      }),
+      body: enc.encode("👻👻"),
+    });
+  });
+
+  describe("file uploads", function () {
+    it("can upload a File object", async function () {
+      if (typeof File === "undefined") {
+        this.skip();
+      }
+
+      const result = await performRequest({
+        file: new File([new Uint8Array([1, 2, 3])], "file.bin", {
+          type: "application/octet-stream",
+        }),
+      });
+
+      const parts = (result.request.multipartBody as MultipartRequestBody).parts;
+      assert.ok(parts.length === 1, "expected 1 part");
+      assert.deepEqual(
+        parts[0].headers,
+        createHttpHeaders({
+          "Content-Type": "application/octet-stream",
+          "Content-Disposition": `form-data; name="file"; filename="file.bin"`,
+        }),
+      );
+      const buf = new Uint8Array(await new Response((parts[0].body as any).stream()).arrayBuffer());
+      assert.deepEqual([...buf], [1, 2, 3]);
+    });
+
+    it("can upload a Blob object", async function () {
+      if (typeof Blob === "undefined") {
+        this.skip();
+      }
+
+      const result = await performRequest({
+        file: new Blob([new Uint8Array([1, 2, 3])]),
+      });
+
+      const parts = (result.request.multipartBody as MultipartRequestBody).parts;
+      assert.ok(parts.length === 1, "expected 1 part");
+      assert.deepEqual(
+        parts[0].headers,
+        createHttpHeaders({
+          // Content-Type should not be inferred
+          "Content-Disposition": `form-data; name="file"; filename="blob"`,
+        }),
+      );
+      const buf = new Uint8Array(await new Response((parts[0].body as any).stream()).arrayBuffer());
+      assert.deepEqual([...buf], [1, 2, 3]);
+    });
+
+    it("can upload a Uint8Array using createFile", async function () {
```

Comment:
Could we have a similar test for the browser case?

Reply:
Good point. We will also need to update `createFile` to create a proper `File` object in the browser. I don't think `fetch` will like the shim that we create. Due to the lack of streaming support as discovered in #28102 I think this may involve loading the stream into memory in the `createFileFromStream` case :/